lib/msf/core/post/windows/process.rb

```diff
@@ -44,20 +44,33 @@ def get_notepad_pathname(bits, windir, client_arch)
   end
 
   #
-  # Injects a reflective DLL in to a process, and executes it.
+  # Injects a reflective DLL into a process, and executes it.
   #
   # @param rdll_path [String] The path to the DLL to inject
   # @param param     [String, Integer, nil] The parameter to pass to the DLL's entry point. If this value is a String
   #   then it will first be written into the process memory and then passed by reference. If the value is an Integer,
   #   then the value will be passed as is. If the value is nil, it'll be passed as a NULL pointer.
-  # @param pid       [Integer] The process ID to inject to, if unspecified, a new instance of notepad.exe will be
-  #   launched to host the injected DLL.
+  # @param pid       [Integer] The process ID to inject to, if unspecified, a new instance of a random EXE from the
+  #   process_list array will be launched to host the injected DLL.
   def execute_dll(rdll_path, param=nil, pid=nil)
+    process_list = ['msiexec', 'netsh', 'userinit']
     if pid.nil?
-      print_status('Launching notepad to host the DLL...')
-      notepad_process = client.sys.process.execute('notepad.exe', nil, { 'Hidden' => true })
+      # Get a random process from the process list to spawn.
+      process_cmd = process_list.sample
+
+      # Use Rex's PeParsey as per Spencer's suggestion to determine the true architecture of the DLL we are injecting.
+      pe = Rex::PeParsey::Pe.new_from_file(rdll_path, true)
+      arch = pe.hdr.file['Machine'].value
+
+      # If the DLL is x86 but the host architecture is x64, then launch a 32 bit WoW64 binary to inject into.
+      if (arch == Rex::PeParsey::PeBase::IMAGE_FILE_MACHINE_I386) && (session.arch == ARCH_x64)
```

Comment:
A 32-bit meterpreter running on a 64-bit host will have a `session.arch` of `ARCH_X86` which wouldn't work for this check. Instead the sysinfo hash's `Architecture` key should be used like `session.sys.config.sysinfo['Architecture'] == ARCH_X64`.

Reply:
Resolved with https://github.com/rapid7/metasploit-framework/pull/15384/commits/4ab1b6ca11cc7b025da6094ec4552eaaa8ad657b